### DISCUSSION TITLE
Redirecting to user login page

### DISCUSSION THREAD

**Discussion by odescopi** [danirus/django-comments-xtd#265] [Q&A]
Hi Dan I have successfully changed the login redirects from admin to user page but it seems after logging in as a user the content doesn't render on the same page rather it redirects to a new comment page..is there a work around to render the same article details page even after a user has logged in..so as to carry out your tasks just like it was if you are authenticated..

**Comment by danirus** [comment]
If you were in the URL `/articles/the-title-to-my-story`, and you wanted to go back to that URL after login in, you have to pass the following `next` query string parameter: `?next=/articles/the-title-to-my-story`. Read about it [here](https://docs.djangoproject.com/en/3.1/topics/auth/default/#limiting-access-to-logged-in-users).

**Reply by odescopi** [reply]
Please dan where do i pass these parameters is it in the django comments xtd app views or my users app or the articles

**Comment by odescopi** [comment]
Thanks dan..i dont quite understan what app to fix those parameters for the
redirect on the same page..cos i have django comments xtd app,articles
app,and users app..i posted the question on the platform too

**Reply by danirus** [reply]
I suppose you have to do it in the templates where you display the login link.
If only registered users can post comments, I figure you added something like:

    {% if request.user.is_authenticated %}
      ... here the comment form ...
    {% else %}
        <a href="{% url 'login' %}?next={{ object.get_absolute_url }} ">Login to post comments.</a>
    {% endif %}

The `object` would be the article you want to send the comments against.
I suggest you to read the complete page I linked you above.

**Reply by danirus** [reply]
In addition to that, for your use case (only registered users can comment) I wrote a page in the docs:
  * https://django-comments-xtd.readthedocs.io/en/latest/usecases/only-signed-in-can-comment.html
